docs/timezone-draft.md

```diff
@@ -0,0 +1,180 @@
+Or just replace the `Temporal.TimeZone` class and `Temporal.now` object altogether:
+
+```javascript
+// For example, to allow only offset time zones:
+
+TemporalTimeZone_idToTimeZone = Temporal.TimeZone.idToTimeZone;
+Temporal.TimeZone.idToTimeZone = function (id) {
+  if (/^[+-]\d{2}:?\d{2}$/.test(id) || id === 'UTC')
+    return TemporalTimeZone_idToTimeZone(id);
+  return null;
+}
+Temporal.TimeZone[Symbol.iterator] = function* () { return null; }
+Temporal.now.timeZone = function () { return Temporal.TimeZone.from('UTC'); }
+
+// or, to replace the built-in implementation altogether:
+
+Temporal.TimeZone = LockedDownTimeZoneImplementation;
+Temporal.now = lockedDownNowObject;
+```
+
+## Implementation of a custom time zone
+
+### TimeZone interface
+
+```javascript
+class Temporal.TimeZone {
+  /** Sets the [[Identifier]] internal slot to @id, and creates the
+   * [[InitializedTemporalTimeZone]] internal slot. A subclassed custom
+   * time zone must chain up to this constructor. */
+  constructor(id : string) : Temporal.TimeZone;
+
+  // Methods that a subclassed custom time zone must implement
+
+  /** Given an absolute instant returns this time zone's corresponding
+   * UTC offset, in nanoseconds (signed). */
+  getOffsetAtInstant(absolute : Temporal.Absolute) : number;
+
+  /** Given the calendar/wall-clock time, returns an array of 0, 1, or
+   * 2 absolute instants that are possible points on the timeline
+   * corresponding to it. In getAbsoluteFor(), one of these will be
+   * selected, depending on the disambiguation option. */
+  possibleInstants(dateTime : Temporal.DateTime) : array<Temporal.Absolute>;
+
+  /** Returns an iterator of all following offset transitions, starting
+   * from @startingPoint. */
+  *getTransitions(startingPoint : Temporal.Absolute) : iterator<Temporal.Absolute>;
+
+  // API methods that a subclassed custom time zone doesn't need to touch
+
+  get name() : string;
+  getDateTimeFor(absolute : Temporal.Absolute) : Temporal.DateTime;
+  getAbsoluteFor(
+      dateTime : Temporal.DateTime,
+      options?: object
+  ) : Temporal.Absolute;
+  getOffsetFor(absolute : Temporal.Absolute) : string;
+  toString() : string;
+  toJSON() : string;
+
+  static from(item : any, options?: object) : Temporal.TimeZone;
+  static idToTimeZone(id: string) : Temporal.TimeZone;
+  static [Symbol.iterator]() : iterator<Temporal.TimeZone>;
+}
+```
+
+All the methods that custom time zones inherit from `Temporal.TimeZone` are implemented in terms of `getOffsetAtInstant()`, `possibleInstants()`, and the value of the _[[Identifier]]_ internal slot.
+For example, `getOffsetFor()` and `getDateTimeFor()` call `getOffsetAtInstant()`, and `getAbsoluteFor()` calls both.
+
+> **FIXME:** These names are not very good.
+> Help is welcome in determining the color of this bike shed.
+
+Alternatively, a custom time zone doesn't have to be a subclass of `Temporal.TimeZone`.
```

Comment:
Another question: what happens if you call `Temporal.TimeZone.from()` with such a plain object?

Reply:
Since the plain object must implement the `name` property, it would be the same result as calling `Temporal.TimeZone.from(obj.name)`.

Reply:
Hmm, can we require `toString()`? Then `from()` will call it by default. But then again, you'll still need the `idToTimeZone`…

Reply:
Yes, that would be fine, implementing `name` would be optional then. I'll change this.

In #294 various ways of getting around the `idToTimeZone` requirement are suggested.